layer/runtime_layer.cc

```diff
@@ -169,41 +169,78 @@ VKAPI_ATTR void RuntimeLayer_CmdDispatch(VkCommandBuffer command_buffer,
                                          uint32_t group_count_y,
                                          uint32_t group_count_z) {
   performancelayers::RuntimeLayerData* layer_data = GetLayerData();
-
-  VkQueryPool timestamp_query_pool =
-      layer_data->GetNewTimeStampQueryPool(command_buffer);
   VkDevice device = layer_data->GetDevice(command_buffer);
-  auto write_timestamp_function = layer_data->GetNextDeviceProcAddr(
-      device, &VkLayerDispatchTable::CmdWriteTimestamp);
   auto next_proc = layer_data->GetNextDeviceProcAddr(
       device, &VkLayerDispatchTable::CmdDispatch);
 
-  // Get the timestamp before the dispatch starts.
-  (write_timestamp_function)(command_buffer,
-                             VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
-                             timestamp_query_pool, 0);
+  VkQueryPool timestamp_query_pool = VK_NULL_HANDLE;
+  VkQueryPool stat_query_pool = VK_NULL_HANDLE;
+  if (!layer_data->GetNewQueryInfo(command_buffer, &timestamp_query_pool,
```

Comment:
when can we get a null here?

Reply:
IE, can you add a comment saying when this can happen and why it's fine to just bail out

Reply:
done